goon/code/datums/browserOutput.dm

```diff
@@ -221,7 +221,7 @@ For the main html chat area
 
 	// Either an atom or somebody fucked up and is gonna get a runtime, which I'm fine with.
 	var/atom/A = obj
-	var/key = "[istype(A.icon, /icon) ? "\ref[A.icon]" : A.icon]:[A.icon_state]"
+	var/key = ("[A.icon]" || "\ref[A.icon]")+":[A.icon_state]"
 	if (!bicon_cache[key]) // Doesn't exist, make it.
 		var/icon/I = icon(A.icon, A.icon_state, SOUTH, 1)
 		if (ishuman(obj)) // Shitty workaround for a BYOND issue.
```

Comment:
Oh I didn't realize how close this was to the other thing
While you're at it, could you replace this `ishuman()` check with `!"[A.icon]"`? I know it sounds INTENSELY stupid, but it's actually the proper way to do this.
Doing so will fix an unreported issue which makes pAIs and possibly some other things appear as four directional sprites rather than one